@codex please fix

```
Run python -m pytest -q tests/e2e/test_ui.py -k 'landing_chat_real_inference_with_desktop_bridge_api_v1'
============================= test session starts ==============================
platform linux -- Python 3.11.15, pytest-8.4.2, pluggy-1.6.0
benchmark: 5.1.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/runner/work/token.place/token.place
configfile: pytest.ini
plugins: base-url-2.1.0, playwright-0.7.1, cov-7.1.0, mock-3.15.1, anyio-4.13.0, benchmark-5.1.0
collected 7 items / 6 deselected / 1 selected

tests/e2e/test_ui.py::test_landing_chat_real_inference_with_desktop_bridge_api_v1 FAILED [100%]

=================================== FAILURES ===================================
_________ test_landing_chat_real_inference_with_desktop_bridge_api_v1 __________

page = <Page url='http://localhost:5010/'>, base_url = 'http://localhost:5010'
setup_servers = (<Popen: returncode: None args: ['/opt/hostedtoolcache/Python/3.11.15/x64/bin...>, None)

    @pytest.mark.e2e
    def test_landing_chat_real_inference_with_desktop_bridge_api_v1(
        page: Page,
        base_url: str,
        setup_servers,
    ):
        """
        Validate relay landing chat end-to-end through the desktop compute-node bridge.
    
        This test intentionally avoids route mocking so it verifies the real wiring:
        browser UI -> relay.py API v1 -> relay sink/source -> desktop bridge runtime.
        """
        relay_process, _ = setup_servers
        assert relay_process is not None
    
        test_env = os.environ.copy()
        test_env["TOKEN_PLACE_ENV"] = "testing"
        test_env["USE_MOCK_LLM"] = "0"
        preprovisioned_model_path = os.environ.get("TOKENPLACE_REAL_E2E_MODEL_PATH", "").strip()
        if not preprovisioned_model_path:
            raise AssertionError(
self = <playwright._impl._connection.Connection object at 0x7fe3c0ed4810>
cb = <function Channel.send.<locals>.<lambda> at 0x7fe3bfe2e980>
is_internal = False, title = None

    async def wrap_api_call(
        self, cb: Callable[[], Any], is_internal: bool = False, title: str = None
    ) -> Any:
        if self._api_zone.get():
            return await cb()
        task = asyncio.current_task(self._loop)
        st: List[inspect.FrameInfo] = getattr(
            task, "__pw_stack__", None
        ) or inspect.stack(0)
    
        parsed_st = _extract_stack_trace_information_from_stack(st, is_internal, title)
        self._api_zone.set(parsed_st)
        try:
            return await cb()
        except Exception as error:
>           raise rewrite_error(error, f"{parsed_st['apiName']}: {error}") from None
E           playwright._impl._errors.TimeoutError: Locator.wait_for: Timeout 30000ms exceeded.
E           Call log:
E             - waiting for locator(".assistant-message").first to be visible

/opt/hostedtoolcache/Python/3.11.15/x64/lib/python3.11/site-packages/playwright/_impl/_connection.py:559: TimeoutError
---------------------------- Captured stdout setup -----------------------------

Setting up servers for E2E tests...
Started relay server on port 5010 with real-provider guardrails
✓ Relay server is running
Real desktop-bridge guardrail mode: skipping server.py registration bootstrap to avoid non-bridge node selection
----------------------------- Captured stdout call -----------------------------
Browser Console [info]: <redacted>
Browser Console [info]: <redacted>
--------------------------- Captured stdout teardown ---------------------------
Error: Browser Console [error]: <redacted>
=========================== short test summary info ============================
FAILED tests/e2e/test_ui.py::test_landing_chat_real_inference_with_desktop_bridge_api_v1 - playwright._impl._errors.TimeoutError: Locator.wait_for: Timeout 30000ms exceeded.
Call log:
  - waiting for locator(".assistant-message").first to be visible
======================= 1 failed, 6 deselected in 36.84s =======================
Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/relay.py
+++ b/relay.py
@@ -786,13 +786,13 @@ def api_v1_relay_servers_poll():
     first_request = None
     while queued_requests:
         candidate = queued_requests[0]
-        if candidate.get('e2ee_v1'):
+        is_api_v1_envelope = bool(candidate.get('e2ee_v1'))
+        is_legacy_ciphertext_payload = all(
+            key in candidate for key in ('client_public_key', 'chat_history', 'cipherkey', 'iv')
+        )
+        if is_api_v1_envelope or is_legacy_ciphertext_payload:
             first_request = queued_requests.pop(0)
             break
-        LOGGER.warning(
-            "relay.api_v1_legacy_payload_skipped",
-            extra={"server_public_key": public_key},
-        )
         queued_requests.pop(0)
 
     if first_request is None:

--- a/tests/unit/test_relay_client.py
+++ b/tests/unit/test_relay_client.py
@@ -819,6 +819,7 @@ class TestRelayClient:
             'next_ping_in_x_seconds': 7,
         }
 
+    @patch('utils.networking.relay_client.requests.post')
     def test_poll_api_v1_encrypted_work_propagates_register_interval_on_poll_error(self, mock_post, relay_client):
         register_ok = MagicMock(status_code=200)
         register_ok.json.return_value = {'next_ping_in_x_seconds': 11}

--- a/tests/unit/test_relay_client.py
+++ b/tests/unit/test_relay_client.py
@@ -805,6 +805,20 @@ class TestRelayClient:
         ]
 
     @patch('utils.networking.relay_client.requests.post')
+    def test_poll_api_v1_encrypted_work_non_dict_payload_returns_bounded_error(self, mock_post, relay_client):
+        register_ok = MagicMock(status_code=200)
+        register_ok.json.return_value = {'next_ping_in_x_seconds': 7}
+        poll_ok_non_dict = MagicMock(status_code=200)
+        poll_ok_non_dict.json.return_value = ['unexpected']
+        mock_post.side_effect = [register_ok, poll_ok_non_dict]
+
+        result = relay_client.poll_api_v1_encrypted_work()
+
+        assert result == {
+            'error': 'Invalid response format: expected object payload',
+            'next_ping_in_x_seconds': 7,
+        }
+
     def test_poll_api_v1_encrypted_work_propagates_register_interval_on_poll_error(self, mock_post, relay_client):
         register_ok = MagicMock(status_code=200)
         register_ok.json.return_value = {'next_ping_in_x_seconds': 11}

--- a/utils/networking/relay_client.py
+++ b/utils/networking/relay_client.py
@@ -721,8 +721,13 @@ class RelayClient:
                     }
                     continue
                 payload = response.json()
-                if isinstance(payload, dict):
-                    payload.setdefault('next_ping_in_x_seconds', register_wait)
+                if not isinstance(payload, dict):
+                    last_error = {
+                        'error': 'Invalid response format: expected object payload',
+                        'next_ping_in_x_seconds': register_wait,
+                    }
+                    continue
+                payload.setdefault('next_ping_in_x_seconds', register_wait)
                 self._active_relay_index = index
                 return payload
             except Exception as exc:

